tt_metal/impl/dispatch/device_command.hpp

```diff
@@ -253,6 +252,11 @@ class DeviceCommand {
             TT_ASSERT(data != nullptr);  // compiled out?
             uint32_t increment_sizeB = align(data_sizeB, this->pcie_alignment);
             this->add_data(data, data_sizeB, increment_sizeB);
+        } else if constexpr (not flush_prefetch) {
+            // Need to make sure next command that flushes prefetch is written to correctly aligned location
+            constexpr uint32_t cmd_sizes = sizeof(CQPrefetchCmd) + sizeof(CQDispatchCmd);
+            uint32_t increment_sizeB = align(cmd_sizes, this->pcie_alignment) - cmd_sizes;
+            this->cmd_write_offsetB += increment_sizeB;
```

Comment:
Is the inline_data branch also bugged?
It aligns data_sizeB to pcie alignment, but this assumes prefetch + dispatch cmd size is pcie aligned as well right?

Should we just do `this->cmd_write_offsetB = align(this->cmd_write_offsetB, this->pcie_alignment);` at the end of this fn similar to some of the others instead?

Reply:
do we expect to have inline data where `flush_prefetch` is false?

Reply:
hm wait this just modifies the stride, we don't actually increment the cmd_write_offsetB. So Ill do the aligned increment regardless of inline data or not

Reply:
Right, I think I missed the 1 b) case when I suggested to move it outside the if. Seems like it should be in the if, and the inline_data branch needs to be updated to align the entire thing instead of just the data instead